angular2-jwt.ts

```diff
@@ -15,46 +15,41 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+class AuthConfigConsts {
+    public static readonly DEFAULT_TOKEN_NAME = 'id_token';
+    public static readonly DEFAULT_HEADER_NAME = 'Authorization';
+    public static readonly HEADER_PREFIX_BEARER = 'Bearer ';
+}
+
 /**
  * Sets up the authentication configuration.
  */
 
 export class AuthConfig {
-  
-  headerName: string;
-  headerPrefix: string;
-  tokenName: string;
-  tokenGetter: any;
-  noJwtError: boolean;
-  noTokenScheme: boolean;
-  globalHeaders: Array<Object>;
 
-  constructor(config:any={}) {
-    this.headerName = config.headerName || 'Authorization';
-    if (config.headerPrefix) {
-      this.headerPrefix = config.headerPrefix + ' ';
-    } else if (config.noTokenScheme) {
-      this.headerPrefix = '';
+  private _config : IAuthConfig;
+
+  constructor(config:IAuthConfig = {
```

Comment:
Hmm... you are probably right. Thanks for catching that. :rose: 
And making all props of IAuthConfig optional is not a viable solution either, I guess!?

What do you think of introducing another interface (like `IAuthConfigOptional`) whose
properties are all optional? This would then only be used as an argument for the ctor.
Of course we have to cast, when we assign it to `IAuthConfig`.
In order to clarify, here is an example:
```typescript
interface IConfig {
	foo: Number;
	bar: string;
}

interface IConfigOptional {
	foo?: Number;
	bar?: string;
}

class FooBar {
	_config: IConfig;
	
	constructor(config: IConfigOptional = {
		foo: 2,
		bar: "baz"
	}){
		// we have to cast here, but that is no problem,
		// as long as we have default values for all props
		// in ctor definition
		this._config = <IConfig>config;
	}
	
	
}

// we do not specify all props here
var test = new FooBar({foo:2});